### PR TITLE
Refactor native/cpu/zmath.h

### PR DESCRIPTION
**Pull request by zasdfgbnm**
There is now a `zmath.h` and `zmath_std.h`, where the latter is the copy-paste of the original `zmath.h` and supporting `std::complex`, and `zmath.h` is for supporting `c10::complex`. `zmath_std.h` will be removed eventually.

### CODE REVIEW

**Comment by anjali411** [comment]
why do we need `zmath` with c10::complex? Correct me if I am wrong, but isn't `zmath` just used because we need thrust::complex on cuda instead of std::complex which was the defined ctype for complex earlier

**Comment by zasdfgbnm** [comment]
@anjali411 There is a `native/cuda/zmath.h` and `native/cpu/zmath.h`, the cuda zmath is just for wrapping `std::complex` into `thrust::complex`, but on cpu zmath, it has more: for example, it contains some helper such as `zabs` which is used to wrap the `scalar_t` in the implementation `torch.min`.

Although I doubt that `torch.min` is taking the min of complex according to its norm does not make much sense, it is still good to have this refactor to unblock the migration of dispatch macros, and take care of the semantics of operators as a separate task.

File: aten/src/ATen/native/cpu/zmath.h [hunk 1]
```diff
@@ -79,22 +64,22 @@ constexpr VALUE_TYPE real_impl (SCALAR_TYPE z) {
...
 template<>
-constexpr float real_impl <std::complex<float>, float> (std::complex<float> z) {
+constexpr float real_impl <c10::complex<float>, float> (c10::complex<float> z) {
   return z.real();
 }
 
 template<>
-constexpr std::complex<double> real_impl <std::complex<double>> (std::complex<double> z) {
-  return std::complex<double>(z.real(), 0.0);
+constexpr c10::complex<double> real_impl <c10::complex<double>> (c10::complex<double> z) {
```
Comment (zasdfgbnm) [inline aten/src/ATen/native/cpu/zmath.h:77]: Looks like this is not needed, `std::real` should work. @anjali411 Please hold on and I will let you know when this is ready.

File: aten/src/ATen/native/cpu/zmath.h [hunk 2]
```diff
@@ -1,50 +1,35 @@
...
-
 template <typename SCALAR_TYPE, typename VALUE_TYPE=SCALAR_TYPE>
 inline VALUE_TYPE zabs (SCALAR_TYPE z) {
   return z;
 }
 
 template<>
-inline std::complex<float> zabs <std::complex<float>> (std::complex<float> z) {
-  return std::complex<float>(std::abs(z));
+inline c10::complex<float> zabs <c10::complex<float>> (c10::complex<float> z) {
```
Comment (zasdfgbnm) [inline aten/src/ATen/native/cpu/zmath.h:17]: This should be still needed.

**Review by anjali411 — APPROVED** [review]

**Pull request closed without merging**